docs/reference/aggregations/bucket/categorize-text-aggregation.asciidoc

```diff
@@ -0,0 +1,446 @@
+[[search-aggregations-bucket-categorize-text-aggregation]]
+=== Categorize text aggregation
+++++
+<titleabbrev>Categorize text</titleabbrev>
+++++
+
+experimental::[]
+
+A multi-bucket aggregation that groups semi-structured text into buckets. Each `text` field is re-analyzed
+using a custom analyzer. The resulting tokens are then categorized creating buckets of similarly formatted
+text values. This aggregation works best with machine generated text like system logs.
+
+WARNING: Re-analyzing _large_ result sets will require a lot of time and memory. This aggregation should be
+         used in conjunction with <<async-search, Async search>>.
+
+[[bucket-categorize-text-agg-syntax]]
+==== Parameters
+
+`field`::
+(Required, string)
+The semi-structured text field to categorize.
+
+`max_children`::
+(Optional, integer, default: `50`)
+The maximum number of unique tokens at any given layer of the tokenization tree.
+Must be larger than 1. Smaller values use less memory and create fewer categories.
+Larger values will use more memory and create more categories.
+
+`max_depth`::
+(Optional, integer, default: `5`)
+The maximum number of tokens matched on before attempting to merge categories.
+Larger values will use more memory and create narrower categories.
```

Comment:
`max_tokens` would be tricky as well as it may assume that "this is the max size log we match", which isn't true. Its just the number of token vectors.

I am still thinking on a better name